ai2thor_wrapper/envs.py

```diff
@@ -0,0 +1,195 @@
+                        break
+        elif self.ACTION_SPACE[action_int]['action'] == 'PutObject':
+            if len(self.event.metadata['inventoryObjects']) > 0:
+                for o in self.event.metadata['objects']:
+                    # loop through receptacles
+                    if o['visible'] and o['receptacle'] and \
+                            o['objectType'] in self.receptacle_object_types and \
+                            len(o['receptacleObjectIds']) < o['receptacleCount']:
+                        # todo might still crash.
+                        inventory_object_id = self.event.metadata['inventoryObjects'][0]['objectId']
+                        inventory_object_type = self.event.metadata['inventoryObjects'][0]['objectType']
+
+                        self.event = self.controller.step(dict(action='PutObject', objectId=inventory_object_id,
+                                                               receptacleObjectId=o['objectId']))#,
+                                                               #raise_for_failure=True)
+                        if inventory_object_type == self.current_object_type:
+                            self.goal_objects_collected_and_placed.remove(inventory_object_id)
+                        print('Placed', inventory_object_id, ' onto', o['objectId'], ' Inventory: ',
+                              self.event.metadata['inventoryObjects'])
+                        break
+        elif self.ACTION_SPACE[action_int]['action'] == 'OpenObject':
+            for o in self.event.metadata['objects']:
+                # loop through objects that are visible, openable, closed
+                if o['visible'] and o['openable'] and not o['isopen'] and o['objectType'] == 'Microwave' \
+                        and o['objectId'] in self.event.instance_detections2D:
+                    print('Opened', o['objectId'])
+
+                    self.event = self.controller.step(
+                        dict(action='OpenObject', objectId=o['objectId']), raise_for_failure=True)
+                    break
+        elif self.ACTION_SPACE[action_int]['action'] == 'CloseObject':
+            for o in self.event.metadata['objects']:
+                # loop through objects that are visible, openable, open
+                if o['visible'] and o['openable'] and o['isopen'] and o['objectType'] == 'Microwave' \
+                        and o['objectId'] in self.event.instance_detections2D:
+                    print('Closed', o['objectId'])
+                    self.event = self.controller.step(
+                        dict(action='CloseObject', objectId=o['objectId']), raise_for_failure=True)
+                    break
+        else:
+            action = self.ACTION_SPACE[action_int]
+            self.event = self.controller.step(action)
+
+        self.t += 1
+        state = self.preprocess(self.event.frame)
+        reward = self.calculate_reward(self.done)
+        self.done = self.is_episode_finished()
```

Comment:
Please do that in the function name as well
```suggestion
        self.done = self.episode_finished()
```

Reply:
yeah, i hated that function name